protocols/v2/roles-logic-sv2/src/utils.rs

```diff
@@ -1331,4 +1408,59 @@ mod tests {
         // m.super_safe_lock(|i| *i = (*i).checked_add(1).unwrap()); // will not compile
         m.super_safe_lock(|i| *i = (*i).checked_add(1).unwrap_or_default()); // compiles
     }
+
+    #[test]
+    fn test_reconstruct_with_bip141() {
```

Comment:
It feels like the new code is testing itself here because you are comparing if the results from `Coinbase::new` and `Coinbase::reconstruct_from..` are similar